internal/extender/resource.go

```diff
@@ -193,6 +193,21 @@ func (s *SparkSchedulerExtender) fitEarlierDrivers(
 }
 
 func (s *SparkSchedulerExtender) selectDriverNode(ctx context.Context, driver *v1.Pod, nodeNames []string) (string, string, error) {
+	if rr, ok := s.resourceReservations.Get(driver.Namespace, driver.Labels[SparkAppIDLabel]); ok {
+		driverReservedNode := rr.Spec.Reservations["driver"].Node
+		for _, node := range nodeNames {
+			if driverReservedNode == node {
+				svc1log.FromContext(ctx).Info("Received request to schedule driver which already has a reservation. Returning previously reserved node...",
+					svc1log.SafeParam("namespace", driver.Namespace),
+					svc1log.SafeParam("driverReservedNode", driverReservedNode))
+				return driverReservedNode, success, nil
+			}
+		}
+		svc1log.FromContext(ctx).Warn("Received request to schedule driver which already has a reservation, but previously reserved node is not in list of nodes",
```

Comment:
I think we should log Warn, but still return the driverReservedNode here to preserve the previous behaviour

Reply:
Addressed for consistency with previous behavior, but what is `kube-scheduler`'s behavior in this case?

Reply:
I think currently it doesn't check filter result, so extenders can return nodes that were not part of the input list.
https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/core/generic_scheduler.go#L530-L547